primitives/runtime/src/traits.rs

```diff
@@ -1335,26 +1335,29 @@ pub trait GetNodeBlockType {
 	type NodeBlock: self::Block;
 }
 
-/// Something that can validate unsigned extrinsics for the transaction pool.
+/// Provide validation for unsigned extrinsics.
 ///
-/// Note that any checks done here are only used for determining the validity of
-/// the transaction for the transaction pool.
-/// During block execution phase one need to perform the same checks anyway,
-/// since this function is not being called.
+/// This trait provides two functions [`pre_dispatch`](Self::pre_dispatch) and
+/// [`validate_unsigned`](Self::validate_unsigned). The first one being called right before
```

Comment:
```suggestion
/// [`validate_unsigned`](Self::validate_unsigned). The `pre_dispatch` function is called before
```
If you want to emphasize "right" before, I'd suggest using "immediately" but it's fine to leave it out.